src/AppBundle/Controller/DefaultController.php

```diff
@@ -0,0 +1,172 @@
+            $form['file']->getData()->move('/tmp', 'example.tmp');
+
+            $lines = explode(PHP_EOL, file_get_contents('/tmp/example.tmp'));
+
+            // since we don't care about first line or last line
+            array_shift($lines);
+            array_pop($lines);
+
+            foreach ($lines as $line) {
+                $data = explode("\t", $line);
+
+                $product = new Product();
+                $product->setTitle($data[0]);
+                $product->setDescription($this->sanitize($data[1]));
+                $product->setPrice($data[2]);
+
+                // I'm not dealing with timezones as I'd need another field for it to work with doctrine and I didn't notice until the end of the exam
+                $product->setInitDate(new \DateTime($data[3]));
+                $product->setExpiryDate(new \DateTime($data[4]));
+
+                $product->setMerchantAddress($data[5]);
+                $product->setMerchantName($this->sanitize($data[6]));
+
+                // I'm currently just enabling all of them
+                $product->setStatus(1);
+
+                $this->saveProduct($product);
+            }
+        }
+
+        return $this->render('default/success.html.twig');
+    }
+
+    /**
+     * @Route("/success", name="success")
+     */
+    public function successAction(Request $request)
+    {
+        $months = array();
+
+        foreach ($this->getActiveProducts() as $product) {
+            for ($year = $product['year_start']; $year <= $product['year_end']; ++$year) {
+		// after sleeping I just realized theres a bug here that I'm not going to fix so you can see the error
+		// when the year changes the $month should become 1 and go until 12 until $month = $month_end and $year = $year_end
+		// this way it gets the full range of when the product applys, my mistake for not testing properly
+                for ($month = $product['month_start']; $month <= $product['month_end']; ++$month) {
+                    $months[$year][$month] = !empty($months[$year][$month]) ? $months[$year][$month] + 1 : 1;
+                }
+            }
+        }
+
+        $merchants = $this->getMerchantProducts();
+
+        return $this->render('default/success.html.twig', array('months' => $months, 'merchants' => $merchants));
+    }
+
+    // Could be put in services
+    private function sanitize(&$string)
+    {
+        return trim($string, '""');
+    }
+
+    private function saveProduct($product)
+    {
+        $em = $this->getDoctrine()->getManager();
+        $em->persist($product);
+        $em->flush();
+    }
+
+    // Could be in a repo
+    private function getActiveProducts()
+    {
+        // raw query as requested, get ranges of months for post calculation
+        $sql = "
+          SELECT MONTH(a.init_date) month_start, MONTH(a.expiry_date) month_end, YEAR(a.init_date) year_start, YEAR(a.expiry_date) year_end
+          FROM product a
+          INNER JOIN months_in_year months ON
```

Comment:
After sleeping and a quick revision: this join in the query is actually useless, it was originally made for a complex query but since I post process it in PHP it makes no sense.